commercial/app/model/commercial/events/LiveEventMembershipInfo.scala

```diff
@@ -0,0 +1,22 @@
+package model.commercial.events
+
+import play.api.data.validation.ValidationError
+import play.api.libs.json._
+
+case class LiveEventMembershipInfo(id: String,
+                                   url: String,
+                                   mainImageUrl: String)
+
+object LiveEventMembershipInfo {
+  implicit val format = Json.format[LiveEventMembershipInfo]
+
+  // based on play.api.libs.json.LowPriorityDefaultReads.traversableReads
+  implicit val formats = new Reads[Seq[LiveEventMembershipInfo]] {
+    override def reads(json: JsValue): JsResult[Seq[LiveEventMembershipInfo]] = {
+      json match {
+        case JsArray(jsValues) => JsSuccess(jsValues.flatMap(_.asOpt[LiveEventMembershipInfo]))
+        case _ => JsError(Seq(JsPath() -> Seq(ValidationError("error.expected.jsarray containing LiveEventMembershipInfo"))))
```

Comment:
I'm not sure - it is definitely worth trying - but if I recall correctly, the macros have no tolerance for elements within the `JsArray` that are malformed and would throw a `JsError` only, hence this approach of mapping and manually transforming and flattening each one.